ert3/storage/_storage.py

```diff
@@ -253,17 +306,46 @@ def _get_numerical_data(
     return ert3.data.EnsembleRecord(records=records)
 
 
+def _get_experiment_parameters(
+    workspace: Path, experiment_name: str
+) -> Mapping[str, Iterable[str]]:
+    experiment = _get_experiment_by_name(experiment_name)
+    if experiment is None:
+        raise ert3.exceptions.NonExistantExperiment(
+            f"Cannot get parameters from non-existing experiment: {experiment_name}"
+        )
+
+    ensemble_id = experiment["ensemble_ids"][0]  # currently just one ens per exp
+    response = requests.get(url=f"{_STORAGE_URL}/ensembles/{ensemble_id}/parameters")
+    if response.status_code != 200:
+        raise ert3.exceptions.StorageError(response.text)
+    parameters = {}
+    for name in response.json():
+        key, val = name.split(".")
+        if key in parameters:
+            parameters[key].append(val)
+        else:
+            parameters[key] = [val]
+    return parameters
+
+
 def add_ensemble_record(
     *,
     workspace: Path,
     record_name: str,
     ensemble_record: ert3.data.EnsembleRecord,
     experiment_name: Optional[str] = None,
+    is_parameter: bool = False,
 ) -> None:
     if experiment_name is None:
         experiment_name = f"{workspace}.{_ENSEMBLE_RECORDS}"
 
-    _add_numerical_data(workspace, experiment_name, record_name, ensemble_record)
+    if is_parameter:
```

Comment:
Would be nice to see whether we could unify the code path for parameters and not parameters a bit more. But that can be left as a refactoring exercise when everything is working :)

Reply:
The way we upload parameters and non-parameters is big in part due to the concrete issue this PR is solving, and https://github.com/equinor/ert-storage/issues/126 .